src/install.ml

```diff
@@ -49,34 +64,50 @@ module Section = struct
       ]
 
   module Paths = struct
-    let lib         = Path.in_source "lib"
-    let libexec     = Path.in_source "lib"
-    let bin         = Path.in_source "bin"
-    let sbin        = Path.in_source "sbin"
-    let toplevel    = Path.in_source "lib/toplevel"
-    let share       = Path.in_source "share"
-    let share_root  = Path.in_source "share_root"
-    let etc         = Path.in_source "etc"
-    let doc         = Path.in_source "doc"
-    let stublibs    = Path.in_source "lib/stublibs"
-    let man         = Path.in_source "man"
-  end
+    type t =
+      { lib         : Path.t
+      ; libexec     : Path.t
+      ; bin         : Path.t
+      ; sbin        : Path.t
+      ; toplevel    : Path.t
+      ; share       : Path.t
+      ; share_root  : Path.t
+      ; etc         : Path.t
+      ; doc         : Path.t
+      ; stublibs    : Path.t
+      ; man         : Path.t
+      }
+
+    let make ~package ~destdir ?(libdir=Path.relative destdir "lib") () =
+      let package = Package.Name.to_string package in
+      { bin        = Path.relative destdir "bin"
+      ; sbin       = Path.relative destdir "sbin"
+      ; toplevel   = Path.relative libdir  "toplevel"
+      ; share_root = Path.relative libdir  "share"
+      ; stublibs   = Path.relative libdir  "lib/stublibs"
+      ; man        = Path.relative destdir "man"
+      ; lib        = Path.relative libdir  package
+      ; libexec    = Path.relative libdir  package
+      ; share      = Path.relative destdir ("share/" ^ package)
```

Comment:
Is it OK to concatenate paths like this? I guess `package` is normalized already (so it does not contain slashes or `".."`), but also does it work on windows?

Reply:
There might be an issue here. However we were doing the same before and we have had no issue so far.  I guess we could add a few more constraints on package names